### ISSUE TITLE
Protractor tests hanging in Chrome after navigating to initial data url

### ISSUE THREAD

**Issue by diegope** [angular/protractor#1198]
After the recent Chrome update I've been having issues getting my specs to run using ChromeDriver. The test execution stalls after trying to navigate to the first data URL. It seems that the URL gets garbled somehow on its way into the browser, as it ends up as: "data:," in the URL field.

Here is the output from the driver:

A Jasmine spec timed out. Resetting the WebDriver Control Flow.
The last active task was:

```
WebDriver.navigate().to(data:text/html,<html></html>)
```

The chrome driver version is 2.10, and the browser version is 36.0.1985.143

**Comment by jamm** [comment]
I saw very funny reproduction of this issue today: one app running e2e test just fine and another hanging in connection status.
But I figured out quickly enough what was the reason: wrong 'baseUrl' in config ;) And symptoms were exactly as described in this github issue.

**Comment by basu-ks** [comment]
Hi folks,
Facing the same issue, what is the workaround?
Protractor 1.4/1.6
Chrome: Version 39.0.2171.95 (64-bit)
OS: Mac OSx
ChromeDriver 2.12.301326

**Comment by basu-ks** [comment]
protractor is unable to launch app url in any of the browser, it just opens the browser.Not sure what is going on:(

**Comment by jamm** [comment]
@basu-ks, maybe your app should be launched before tests. Maybe something like `gulp serve` or `grunt serve` or `ionic serve`?

**Comment by zaeni** [comment]
Just started learning angular, following the instructions at 
https://docs.angularjs.org/tutorial 
Already did 'npm start' which seems to start it locally and i can access it through chrome, but i'm getting the same "data:," in the url field of chrome which it starts up and not ending up with the local url and it never gets any further. This is  after i did an 'npm start' in one terminal (which, correct me if i'm wrong, is similar to gulp serve, or grunt serve?) which was the instruction in the tutorial, followed by a 'npm run protractor'.

**Comment by hokkaido** [comment]
I also have this issue from time to time, locally on OSX, I haven't found a pattern yet, sometimes my specs work, sometimes not, they hang at:

`Executing: [get: data:text/html,<html></html>])`

**Comment by DennisVis** [comment]
I was also running into this issue and found out it occurred when I used the getText() function on an element within my page object. Doing the same from within the actual test did not produce any errors.

I hope that helps.

**Comment by SET001** [comment]
How to check my Chromedriver version?
I'm experiencing same issue on one of my development environments.

**Comment by flaviomeira** [comment]
I have Protractor tests triggered by Bamboo. It hangs with that url (data:text/html,<html></html>) just on Master branch. The same tests, triggered by the same Bamboo runs correctly on others branches. I can't understand what is happening.
I hope we can find a fast workaround :(

**Comment by underscorebrody** [comment]
Adding that I've seen this happen intermittently on tests running either on Mac or in CircleCI. Here's my setup:

Protractor: 1.8 (has happened on both)
Angular: 1.3
Chrome: 42.0.2311
ChromeDriver: 2.14
NVM: 0.10.33

Note I'm running with `directConnect:true`, and this error does not necessarily occur on the first test, but can also occur deep within our test suite. Here is a bit more info from the stack trace (with IP obscured):

```
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
The last active task was: 
WebDriver.navigate().to(data:text/html,<html></html>)
    at [object Object].webdriver.WebDriver.schedule (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:345:15)
    at [object Object].webdriver.WebDriver.Navigation.to (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:1031:23)
    at [object Object].webdriver.WebDriver.get (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:723:26)
    at [object Object].Protractor.get (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/lib/protractor.js:443:15)
    ...
    at /home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/jasminewd/index.js:94:14
    at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1654:20)
    at [object Object].webdriver.promise.ControlFlow.runEventLoop_ (/home/ubuntu/nvm/v0.10.33/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1518:8)
    at [object Object].wrapper [as _onTimeout] (timers.js:261:14)
    at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)
```

@juliemr let me know if there's any more info i can provide that might help sort this out

**Comment by bkbonner** [comment]
I had something like this happen and it turned out a single line of my test was causing it to fail.   My test wasn't that good, but I'll include it here and show what failed.

Hope this helps someone else.

describe('Test Home Page', function() {
  it('should open edit-profile', function() {
	// this is going to redirect me to a SAML IP (Identity Provider) by doing a redirect
	browser.driver.get('https://localhost:8443/app');
    browser.driver.sleep(1500);

    // these elements are on the redirected website, and after I send ENTER, the SAML IP
    // will redirect me back to my website
    browser.driver.findElement(by.id('username')).sendKeys('your-userid');
    browser.driver.findElement(by.id('password')).sendKeys('your-password');
    browser.driver.findElement(by.id('password')).sendKeys(protractor.Key.ENTER);
    browser.driver.sleep(5000);
    
    // This is an angular site so I understand I'm supposed to access it this way.    
    browser.get('https://localhost:8443/app');
    element(by.css('.user_name .drop_down')).click();
    browser.driver.sleep(1000);
    
    // I want to open up the edit profile menu
    element(by.id('edit-profile')).click();
    // and get the element in the modal that pops up
    var currentPerson = element(by.model('person.name'));
    
    // This next line causes the failure of data:,  in the URL,
    // so I'm commenting it out for now
    // expect(currentPerson).toEqual('MY NAME');
  });
});

**Comment by Discmt** [comment]
Any update on this issue? I've run into similar problems.

**Comment by bkbonner** [comment]
I have not yet.   I just haven't had time to get back to it....been pulled onto other things.

**Comment by Shruti-webonise** [comment]
I am Facing the same issue.
When run my conf.js file, It launches the browser and command prompt.
Test hangs and shows error
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
The last active task was:
WebDriver.manage().deleteAllCookies()

**Comment by sukmad** [comment]
@Shruti-webonise Ru able to resolve? even i am facing same issue. My tests were running fine before, but suddenly i am facing the same issue since today morning

**Comment by Irynn** [comment]
Same problem here. Any solutions found?

**Comment by sukmad** [comment]
Updating the chromedriver resolved the issue. update it to the latest one 2.19, I hope this will resolve your issue

**Comment by 298080208** [comment]
@sukmad Updated chromedriver and it did help,thanks.
And I aslo tried to re-install chrome to an old version to work with the chromedrive I used before,it works well,too.
Guess the latest chrome don't support old version chromedrivers?

**Comment by sukmad** [comment]
Yep. i think Chrome version (44+) required new chromedriver 2.19

**Comment by Gheoan** [comment]
Updating Chromedriver also fixed the problem for me.

**Comment by ShrutiDalvi** [comment]
I was using Protractor 1.8 with updated chrome driver.
I Updated Protractor with 3.0 and issue is resolved.
Thanks.

**Comment by lemelon** [comment]
Maybe there's a "browser.sleep" outside a "it" callback

**Comment by erik-ropez** [comment]
Maybe it could help someone. I'm adding Protractor via NPM.

"node_modules/protractor/bin/protractor" doesn't hang.

"npm run protractor" hangs. package.js scripts has "protractor": "protractor" mapping. "protractor": "node_modules/protractor/bin/protractor" behaves in the same way - hangs.

"npm run protractor protractor.conf.js" doesn't hang.

So it seems you need to pass config file into arguments in order to make it work via "npm run". Even though it's not needed when running protractor directly.

**Comment by dayanamcc** [comment]
having this error, on step 0 from tutorial of protractor.

➜  protract git:(master) ✗ protractor conf.js     
[00:00:04] I/hosted - Using the selenium server at http://localhost:4444/wd/hub
[00:00:04] I/launcher - Running 1 instances of WebDriver
Started
FA Jasmine spec timed out. Resetting the WebDriver Control Flow.


Failures:
1) Protractor Demo App should have a title
  Message:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  Stack:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
        at tryOnTimeout (timers.js:224:11)
        at Timer.listOnTimeout (timers.js:198:5)

1 spec, 1 failure
Finished in 30.023 seconds

on Chrome Web browser its only redirecting to: data:text/html,<html></html>

any solution????

**Comment by ShrutiDalvi** [comment]
@dayanamcc 
Please try using following code in conf.js file.
allScriptsTimeout: 900000

Let me know this resolves your issue.

**Comment by dayanamcc** [comment]
@ShrutiDalvi i tried , allScriptsTimeout: 900000 and getting same error. 
this is what i get: [error](https://i.imgsafe.org/f206d0cd96.png).
 
NOTE: also i tried `return browser.ignoreSynchronization = true;`  at least chrome redirect me to [calculatorURL](http://juliemr.github.io/protractor-demo/) but them nothing more happens and got same [error](https://i.imgsafe.org/f21f82a78d.png).

this example should be simple. but i dont unsderstand the issue

**Comment by kristianmandrup** [comment]
Yes, unfortunately Angular and Protractor are getting old, cranky and dusty... in time we all break down

**Comment by naveen020** [comment]
![error](https://cloud.githubusercontent.com/assets/9438036/20089794/20b7886c-a5af-11e6-8144-1250ccec46bf.PNG)

**Comment by naveen020** [comment]
why i am getting this error ?  
can any one explain how i resolved it....

**Comment by beniamin-kis** [comment]
@naveen020 you need to upgrade your chromedriver. You've got the latest chrome, but not the latest chromedriver (v2.25).
Run "webdriver-manager update". That's gonna fix it.

**Comment by naveen020** [comment]
@beniamin-kis thanq now it is working

**Comment by riddla** [comment]
I'm still encountering this issue with the following (up to date) dependencies:

`protractor` version `4.0.9`
`webdriver-manager` version `10.2.7` 
`chromedriver` version `2.25.426935`
`chrome` version `54.0.2840.87`

For the time being I switched to using [Canary](https://www.google.de/chrome/browser/canary.html) with the following settings within `protractor.conf.js`:

```
capabilities: {
  "browserName": "chrome",
"chromeOptions": {
    "binary": "/Users/XXX/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary"
  }
},
```

**Comment by beniamin-kis** [comment]
latest protractor is 4.0.10. Try that maybe?

**Comment by beniamin-kis** [comment]
I have a different issue, though.
Sometimes my tests are hanging at start with the "data:text/html,<html></html>" url in the browser. And then after 30 seconds or so, the URL changes and my app is open (angular app).
This only happens with Chrome. Firefox loads the page instantly.

**Comment by riddla** [comment]
Right, with protractor `4.0.10` I'm experiencing your issue also: chrome just hangs with the `data:,` URL. (Anyway, thanks for pointing me to the newer protractor version.)

**Comment by bopm** [comment]
Same even with 4.0.11.

**Comment by bopm** [comment]
Seems like this issue: https://bugs.chromium.org/p/chromedriver/issues/detail?id=1552

**Comment by bopm** [comment]
So `export LANG=en_US.UTF-8` fixed that for me. But it's really ugly.

**Comment by simbro** [comment]
I'm having the exact same problem, "localhost can't be reached".  The funny thing is, the tests were working perfectly.  Then without any change whatsoever, spontaneously stopped working.  Getting the exact same error as described by various other people in this thread.

protractor: 4.0.10
chromedriver 2.24

bopm's solution of using `export LANG=en_US.UTF-8` did not fix the issue for me.  I'm getting the same isue on windows, and on Ubuntu with a headless chrome and xvfb.

**Comment by floribon** [comment]
New fix for this issue
-----
I had the same problem of protractor getting stuck, and fixed it by switching to `java 1.7` instead of `java 1.8`, hope that will help.

**Comment by killerinheelzzz** [comment]
I am working with protractor as well, we are using it with our Angular 2 UI Code base. I started working on the tests and I point the tests to Either one of our environments url or a localhost:8000/ url. Since our system is authenticated I launched the browser and get redirected to an Oauth page where I pass in the required credentials. Then the browser is redirected, but it is then redirected to localhost:49155/ and it logs me in as another user even though I have placed in the correct Credentials for another User. 

I have dug through our code base and have not found anything indicating a localhost/49155. I have even made sure that the selenium address is pointing to the localhost/8000 or our designated environment. The base Url is also pointing to the  correct place.

So I am confused as to why is launches me to another localhost and logs me in as another user. When I run localhost locally with it pointing to the correct environment then I do not see these problems , but as soon as I use protractor I see these issues.
